frontend/src/concepts/roleBinding/RoleBindingPermissions.tsx

```diff
@@ -0,0 +1,129 @@
+
+type RoleBindingPermissionsProps = {
+  roleBindingPermissionsRB: ContextResourceData<RoleBindingKind>;
+  permissionOptions?: {
+    type: RoleBindingPermissionsRoleType;
+    description: string;
+  }[];
+  projectName: string;
+  roleRef?: string;
+  labels?: { [key: string]: string };
+  description: React.ReactElement | string;
+  groups: GroupKind[];
+};
+
+const RoleBindingPermissions: React.FC<RoleBindingPermissionsProps> = ({
+  roleBindingPermissionsRB,
+  permissionOptions,
+  projectName,
+  roleRef,
+  labels,
+  description,
+  groups,
+}) => {
+  const {
+    data: roleBindings,
+    loaded,
+    error: loadError,
+    refresh: refreshRB,
+  } = roleBindingPermissionsRB;
+  if (loadError) {
+    return (
+      <EmptyState
+        variant={EmptyStateVariant.lg}
+        data-id="error-empty-state"
+        id={ProjectSectionID.PERMISSIONS}
+      >
+        <EmptyStateHeader
+          titleText="There was an issue loading permissions."
+          icon={<EmptyStateIcon icon={ExclamationCircleIcon} />}
+          headingLevel="h2"
+        />
+        <EmptyStateBody>{loadError.message}</EmptyStateBody>
+      </EmptyState>
+    );
+  }
+
+  if (!loaded) {
+    return (
+      <EmptyState
+        variant={EmptyStateVariant.lg}
+        data-id="loading-empty-state"
+        id={ProjectSectionID.PERMISSIONS}
+      >
+        <Spinner size="xl" />
+        <EmptyStateHeader titleText="Loading" headingLevel="h2" />
+      </EmptyState>
+    );
+  }
+
+  const userTable = (
+    <RoleBindingPermissionsTableSection
+      projectName={projectName}
+      roleRef={roleRef}
+      labels={labels}
+      permissionOptions={permissionOptions}
+      roleBindings={filterRoleBindingSubjects(roleBindings, RoleBindingPermissionsRBType.USER)}
+      roleBindingPermissionsTableType={RoleBindingPermissionsRBType.USER}
+      refresh={refreshRB}
+      typeModifier="user"
+    />
+  );
+
+  const groupTable = (
+    <RoleBindingPermissionsTableSection
+      projectName={projectName}
+      roleRef={roleRef}
+      permissionOptions={permissionOptions}
+      labels={labels}
+      roleBindings={filterRoleBindingSubjects(roleBindings, RoleBindingPermissionsRBType.GROUP)}
+      roleBindingPermissionsTableType={RoleBindingPermissionsRBType.GROUP}
+      refresh={refreshRB}
+      typeAhead={
+        groups.length > 0 ? groups.map((group: GroupKind) => group.metadata.name) : undefined
+      }
+      typeModifier="group"
+    />
+  );
+
+  return (
+    <PageSection
+      isFilled
+      aria-label="project-sharing-page-section"
+      variant="light"
+      id={ProjectSectionID.PERMISSIONS}
+    >
+      <Stack hasGutter>
+        <StackItem>{description}</StackItem>
+        <StackItem>
+          {projectName === MODEL_REGISTRY_DEFAULT_NAMESPACE ? groupTable : userTable}
+        </StackItem>
+        <StackItem>
+          {projectName === MODEL_REGISTRY_DEFAULT_NAMESPACE ? userTable : groupTable}
```

Comment:
I think because this component is under its own `~/concepts/roleBinding` it shouldn't have any logic specific to Model Registry. Can you add a prop like `groupsBeforeUsers: boolean` or something to pass instead?

Reply:
This could be followup tech debt, I'll open an issue.